File: nodejs/src/translationService.ts
import { compositionToFhirBundle, fhirPatientToComposition } from './mappers';
import { OpenEhrComposition, TranslationResult } from './models';

/**
 * Facade selecting the appropriate mapper for an inbound payload. Mirrors the
 * .NET `ITranslationService`. For the demographics proof-of-concept there is a
 * single mapper per direction.
 */
export class TranslationService {
  fhirToOpenEhr(fhirJson: string): TranslationResult<OpenEhrComposition> {
    return fhirPatientToComposition(fhirJson);
  }

  openEhrToFhir(composition: OpenEhrComposition): TranslationResult<string> {
    if (!composition) {
      return { success: false, issues: [{ severity: 'error', message: 'openEHR composition is null.' }] };
    }
    return compositionToFhirBundle(composition);
  }
}

File: java/src/main/java/io/github/hl7ie/fhiropenehrbridge/TranslationService.java
package io.github.hl7ie.fhiropenehrbridge;

import com.fasterxml.jackson.databind.JsonNode;

/** Translation façade mirroring the .NET ITranslationService. */
public final class TranslationService {

    public TranslationResult<JsonNode> fhirToOpenEhr(String fhirJson) {
        return Mappers.fhirPatientToComposition(fhirJson);
    }

    public TranslationResult<JsonNode> openEhrToFhir(JsonNode composition) {
        return Mappers.compositionToFhirBundle(composition);
    }
}
